### PR TITLE
[v0.26.0rc][bugfix] skip generate_mtp_attention_mask_for_decode with sparse attention to avoid hostbound

### PR DESCRIPTION
**Pull request by coder-fny**
GLM5.2 uses sparse attention and therefore doesn't need a mask. However, the current mask generation path has host-side operations that take too long at high concurrency, which makes the MTP stage host-bound and degrades performance.

Fixed issues: #15106 
- vLLM version: v0.26.0
- vLLM main: https://github.com/vllm-project/vllm/commit/d02df748bf9efd99022f1a062597dc3cb3808485

### CODE REVIEW

**Review by pisceskkk — APPROVED** [review]
LGTM, thanks for fixing!
